crates/sargon/src/system/sargon_os/sargon_os.rs

```diff
@@ -334,26 +335,33 @@ impl SargonOS {
         Ok((profile, private_bdfs))
     }
 
-    pub(crate) async fn host_id(&self) -> Result<HostId> {
-        Self::get_host_id(&self.clients).await
-    }
-
-    pub(crate) async fn get_host_id(clients: &Clients) -> Result<HostId> {
+    pub(crate) async fn get_host_id(clients: &Clients) -> HostId {
         debug!("Get Host ID");
         let secure_storage = &clients.secure_storage;
+        let stored_host_id = secure_storage.load_host_id().await;
+        let new_host_id = HostId::generate_new();
```

Comment:
you are calling `HostId::generate_new();` twice, once here, and then once in `Ok(None)` branch. I think best remove this one and put one in the `Err` branch, which means that we wont call `HostId::generate_new();` at all for the `Ok(Some(_))` branch, which is good. because if we e.g. add logging inside `HostId::generate_new();`, it might confuse us to see a new Id being generated even though `match stored_host_id` was `Ok(Some(_))`